--- intermediate_queue.py
+++ intermediate_queue.py
@@ -85,7 +92,7 @@
         Returns:
             List[str]: The job IDs
         """
-        return [job_id.decode() for job_id in self.connection.lrange(self.key, 0, -1)]
+        return [job_id for job_id in self.connection.lrange(self.key, 0, -1)]

     def remove(self, job_id: str) -> None:
         """Removes a job from the intermediate queue.